### ISSUE TITLE
Add PUBLIC_PATH as there is a public directory right now

### ISSUE THREAD

**Issue by sabat24** [koseven/koseven#330]
I migrated from old Kohana to Koseven and I noticed that there is a public directory added. However I couldn't be able to find any variable that I could use to get the path to that directory.
Right now I have got real use cases when it could be helpful
1. DOMPDF requires that all images defined in views should be provided via path not url.
2. Swift requires that every image embedded into body has to be provided with it's path, too.
3. When you try to run some cli command which is related to index.php you have to provide full path to index.php
let's say that you want to start some background task via minion:
`$bgproc = popen('/usr/bin/php ' . PUBLIC_PATH . 'index.php task_runner ' . $task_name.' &', 'r');`

Before you could use DOCROOT as assets and index.php where placed in the root directory. Right now index.php and assets are moved to public. Whats more you can still keep all files in some subdirectory inside public dir.

I think it would be good to define some PUBPATH variable to deal with above cases with the same manner as APPPATH, MODPATH and SYSPATH. Because I could use another directory name for public like public_html

In my opinion there should be one line added to index.php (under `$system = 'system'`)
`$public = 'public_html';`

and then below

```php
// Make the public relative to the docroot, for symlink'd index.php
if ( ! is_dir($public) AND is_dir(DOCROOT.$public))
	$public = DOCROOT.$public;
```

```
define('PUBPATH', realpath($public).DIRECTORY_SEPARATOR);

// Clean up the configuration vars
unset($application, $modules, $system, $public);
```

To my needs I added also full PUBLIC_PATH in my bootstrap.php
```php
Kohana::init(array(
    'base_url'   => '/subdirectory/',
    'base_path' => 'subdirectory'.DIRECTORY_SEPARATOR,  // <- added new variable to kohana::init
));

DEFINE('PUBLIC_PATH', PUBPATH.Kohana::$base_path);
```

As a result:

```
docroot -> full_path\panel\
public_path -> full_path\panel\public_html\subdirectory\
pubpath -> full_path\panel\public_html\
```

Where public_path is an equivalent to absolute url generated by: URL::base(TRUE, FALSE);
Basically base_path could be retreived by replacing slashes with DIRECTORY_SEPARATOR, but I made another value to provide to not call str_replace on each call. $base_path could be also ignored and PUBLIC_PATH could be created directly as there is no need to use kohana::$base_path anywhere else.

**Comment by toitzi** [comment]
Hm using something like `DOCROOT`.'public' was never a problem to me althrough i understand the need if you have multiple calls, but maybe you just define the constant if you need it like you did in your code....not sure if we should add this but i am also not against it. 

What do you say? @neo22s @svenbw @errotan @jstrobel @piotrbaczek ?

**Comment by neo22s** [comment]
@toitzi agree I have created different constants to my needs on index.php 

// Define the absolute paths for configured directories
define('APPPATH', realpath($application).DIRECTORY_SEPARATOR);
define('KOMODPATH', realpath($komodules).DIRECTORY_SEPARATOR);
define('MODPATH', realpath($modules).DIRECTORY_SEPARATOR);
define('SYSPATH', realpath($system).DIRECTORY_SEPARATOR);
define('UPDATESPATH', realpath($updates).DIRECTORY_SEPARATOR);
define('LOGSPATH', realpath($logs).DIRECTORY_SEPARATOR);

I add this when I see I use a lot a variable.

**Comment by sabat24** [comment]
I agree with you about that it's not a big problem. I was thinking more about convention. Basically APPPATH is just DOCROOT.'application', MODPATH it's DOCROOT.'modules' and so on. 
The second constant was just an example how to use PUBPATH in real life example.

**Comment by neo22s** [comment]
I understand and having something maybe like PUBLICPATH will be more clear hehe, adding will not have any cost and can be good, nothing bad about it.

**Comment by svenbw** [comment]
I think it's in line with the other path definitions, so it makes sense to include it in the code.

**Comment by toitzi** [comment]
Ok, `PUBPATH`, `PUBLICPATH`, `SCRIPTPATH`...what naming shall we use.
